CRM/Core/Permission.php

```diff
@@ -525,17 +525,9 @@ public static function checkMenuItem(&$item) {
     }
 
     // if component_id is present, ensure it is enabled
-    if (isset($item['component_id']) && $item['component_id']) {
-      if (!isset(Civi::$statics[__CLASS__]['componentNameId'])) {
-        Civi::$statics[__CLASS__]['componentNameId'] = array_flip(CRM_Core_Component::getComponentIDs());
-      }
-      $componentName = Civi::$statics[__CLASS__]['componentNameId'][$item['component_id']];
-
-      $config = CRM_Core_Config::singleton();
-      if (is_array($config->enableComponents) && in_array($componentName, $config->enableComponents)) {
-        // continue with process
-      }
-      else {
+    if (!empty($item['component_id'])) {
+      $componentName = CRM_Core_Component::getComponentName($item['component_id']);
```

Comment:
It was a wacky `if` statement but note this gives a fatal type error if the component is not enabled.

Reply:
@demeritcowboy I see the problem. `getComponentName` was only returning the name if the component was enabled, otherwise `null`. That actually makes the next line checking `isEnabled` redundant... but I feel like it's bad to rely on that undocumented behavior so I've updated `getComponentName` to always work regardless of the component's enabled status.
All should be fixed now.

Reply:
It scares me a bit because this change assumes everywhere those funcs are used doesn't rely on it returning null. For example it does seem to change PriceSet slightly, but not in a crashy way. I think this is ok.

Reply:
Yea, the old behavior was scary too...
On balance I feel like making this change is slightly less scary than leaving it as it was, but only slightly.

Reply:
FYI @demeritcowboy I did have a skim through usages of that function and I don't see any of them relying on that undocumented behavior for functionality, so that's why I felt confident enough to make the change and why it feels less scary than not making it.